src/utils/event.js

```diff
@@ -0,0 +1,11 @@
+/**
+ * Prevent reloading page without user approval
+ *
+ * @param {Event} e
+ * @return {string}
+ */
+export function handleBeforeUnload(e) {
+  const confirmationMessage = 'Uploaded models are not saved. If you leave this page you will need to upload the model again.';
```

Comment:
Hey Ron, I'm actually not seeing this message displayed in any browser I try on Mac (Chrome, FF, Safari).. have you seen it, and if so on what config?

Reply:
as a matter of fact, the user approval prompt feature when exiting or reloading page can be implemented using `beforeunload` event of windows, no another way, so i think we can't make our own prompt dialog, hope you understand.

Reply:
Yeah, that part I see what you're saying.  I was thinking maybe a dialog before that, but I'm happy with PR as-is for now.